tests/subarray_test.py

```diff
@@ -18,16 +18,112 @@ def subarrays(i):
 def test():
     W.set(3)
 
-    A = dp.ndarray([W, W, W, W])
-    B = dp.ndarray([W, W, W, W])
+    A = dc.ndarray([W, W, W, W])
+    B = dc.ndarray([W, W, W, W])
 
     A[:] = np.mgrid[0:W.get(), 0:W.get(), 0:W.get()]
     for i in range(W.get()):
         A[i, :] += 10 * (i + 1)
-    B[:] = dp.float32(0.0)
+    B[:] = dc.float32(0.0)
 
     subarray(A, B, W=W)
 
 
+def test_strides_propagation_to_tasklet():
+    N = 2
+    vec_stride_W, vec_stride_d0 = (
+        dc.symbol(s) for s in ['vec_stride_W', 'vec_stride_d0']
+    )
+    mat_stride_W, mat_stride_d0, mat_stride_d1 = (
+        dc.symbol(s) for s in ['mat_stride_W', 'mat_stride_d0', 'mat_stride_d1']
+    )
+
+    def build_nsdfg():
+        sdfg = dc.SDFG('vec_to_mat')
+        sdfg.add_symbol('_w', dc.int32)
+        sdfg.add_array(
```

Comment:
I have reverted all changes to code generation and legacy tests. I have found a pattern for array access inside the tasklet that works, although I don't understand it.

@tbennun Access to `_mat` requires all explicit indices, while access to `_vec` only works if I exclude the index for the `_w` slice.